pandas/tests/series/test_constructors.py

```diff
@@ -839,3 +839,27 @@ def test_constructor_cast_object(self):
         s = Series(date_range('1/1/2000', periods=10), dtype=object)
         exp = Series(date_range('1/1/2000', periods=10))
         tm.assert_series_equal(s, exp)
+
+    def test_constructor_generic_timestamp(self):
+        # see gh-15524
+        dtype = np.timedelta64
+        s = Series([], dtype=dtype)
+
+        self.assertTrue(s.empty)
```

Comment:
As a side note, not really a fan of having to rewrite every single test comparison just to be more "idiomatic" with the existing test standard.  Why not write our own in `pandas.util.testing` that we can implement according to idiom and call from everywhere else?

e.g.

~~~python
def assertTrue(val):
    assert val

def assertEqual(a, b):
    assert a == b
~~~